Kernel/Syscalls/thread.cpp

```diff
@@ -55,15 +55,18 @@ ErrorOr<FlatPtr> Process::sys$create_thread(void* (*entry)(void*), Userspace<Sys
 
     auto& regs = thread->regs();
     regs.set_ip((FlatPtr)entry);
-    regs.set_flags(0x0202);
     regs.set_sp(user_sp.value());
-#if ARCH(X86_64)
+
+#if ARCH(I386) || ARCH(X86_64)
+    regs.set_flags(0x0202);
```

Comment:
this marks interrupts as enabled (maybe a helper in thread regs would help here to generlise that properly?)

Reply:
Yep, that could be done in the future, but I'm not here to cleanup x86 code :) I just moved some code around in this commit, and added some more #ifdefs. But this could definitely be improved in some future PR :)